src/components/stimuli/Stimuli.tsx

```diff
@@ -29,15 +41,104 @@ const Stimuli: React.FC = () => {
       }
     }
 
-    fetchStimuliUrls();
-  }, []);
+    if (!isUploading) fetchStimuliUrls();
+  }, [isUploading, file]);
+
+  const fileChangedHandler = (event: any) => {
+    setFileInputText(event.target.files[0].name);
+    setFile(event.target.files[0]);
+  };
+
+  const handleUploadClick = async (e: React.FormEvent) => {
+    e.preventDefault();
```

Comment:
This doesn't appear to be necessary.

Reply:
Prevent default is necessary to prevent page refresh upon submit. We don't get a server response otherwise